src/components/Web3Provider/index.tsx

```diff
@@ -46,7 +47,21 @@ function Updater() {
       provider?.off('debug', trace)
       networkProvider?.off('debug', trace)
     }
-  }, [networkProvider, provider, shouldTrace])
+  }, [analyticsContext, networkProvider, provider, shouldTrace])
+
+  const previousConnectedChainId = usePrevious(chainId)
+  useEffect(() => {
+    const chainChanged = previousConnectedChainId && previousConnectedChainId !== chainId
```

Comment:
do we want to track the initial chain that a user connects with on the first page load? Or only on chain change

Reply:
i think we track that as a user property, and also as a field on the wallet_connected event. so no need to emit this event on page load

Reply:
might be worth adding `previousConnectedChainId` to the event?